### PR TITLE
Backport Migration Option AuthToken

### LINKED ISSUE #13085: Gitlab importer worker crashes
- Gitea version (or commit ref): 1.12.5
- Git version: 2.20.1
- Operating system: Linux/Debian stable
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [X] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [X] No
  - [ ] Not relevant
- Log gist:

## Description

We see some dozen worker crashes seemingly related to the gitlab importer in our logs. Related issue: https://codeberg.org/Codeberg/Community/issues/304

## Logs

```raw
2020/10/09 11:52:43 modules/task/task.go:51:handle() [E] Run task failed: Handler crashed with error: /build/go/src/runtime/panic.go:212 (0x44d709)
        /build/go/src/runtime/signal_unix.go:695 (0x44d558)
        /build/gitea/src/code.gitea.io/gitea/modules/migrations/gitlab.go:126 (0x19089c5)
        /build/gitea/src/code.gitea.io/gitea/modules/migrations/base/downloader.go:58 (0x1872e16)
        /build/gitea/src/code.gitea.io/gitea/modules/migrations/migrate.go:73 (0x190c76d)
        /build/gitea/src/code.gitea.io/gitea/modules/task/migrate.go:99 (0x192042f)
        /build/gitea/src/code.gitea.io/gitea/modules/task/task.go:28 (0x1920c97)
        /build/gitea/src/code.gitea.io/gitea/modules/task/task.go:50 (0x1920f51)
        /build/gitea/src/code.gitea.io/gitea/modules/queue/workerpool.go:383 (0x168de51)
        /build/gitea/src/code.gitea.io/gitea/modules/queue/workerpool.go:238 (0x16901d2)
        /build/go/src/runtime/asm_amd64.s:1373 (0x469720)
```

### PR DESCRIPTION
**Pull request by 6543**
fix #13085

Backport #12862
Backport Parts of #12672

### CODE REVIEW

File: modules/migrations/gitlab.go [hunk 1]
```diff
@@ -85,15 +86,14 @@ type GitlabDownloader struct {
...
+func NewGitlabDownloader(baseURL, repoPath, username, password, token string) *GitlabDownloader {
 	var gitlabClient *gitlab.Client
 	var err error
-	if username != "" {
-		if password == "" {
-			gitlabClient, err = gitlab.NewClient(username, gitlab.WithBaseURL(baseURL))
-		} else {
-			gitlabClient, err = gitlab.NewBasicAuthClient(username, password, gitlab.WithBaseURL(baseURL))
-		}
+	gitlabClient, err = gitlab.NewClient(token, gitlab.WithBaseURL(baseURL))
```
Comment (zeripath) [inline modules/migrations/gitlab.go]: why bother with the var gitlabClient and var error above when you can just
```suggestion
	gitlabClient, err := gitlab.NewClient(token, gitlab.WithBaseURL(baseURL))
```

**Review by zeripath — APPROVED** [review]

File: modules/migrations/gitlab.go (hunk 1 above)
Comment (6543) [reply modules/migrations/gitlab.go]: this is true ... and should be changed upstream too then ...

Comment (6543) [reply modules/migrations/gitlab.go]: this should have been done in #12862 :D

Comment (zeripath) [reply modules/migrations/gitlab.go]: it's not a problem... it can stay in here; just something I noted.

Comment (6543) [reply modules/migrations/gitlab.go]: #13103

**Review by techknowlogick — APPROVED** [review]

**Comment by zeripath** [comment]
Make lg-tm work

**Pull request merged**